examples/java/unit-test/rest/TwilioRestTest.java

```diff
@@ -636,4 +737,139 @@ public void testShouldReadPreviousPage() {
+                com.twilio.rest.Domains.API.toString(),
+                "/2010-04-01/Accounts/AC222222222222222222222222222222.json"
+        );
+        mockRequest.addPostParam("PauseBehavior", "test");
+
+        ObjectMapper objectMapper = new ObjectMapper();
+        objectMapper.registerModule(new JavaTimeModule());
+        when(twilioRestClient.request(Mockito.any())).thenReturn(new Response("{\"account_sid\":\"AC222222222222222222222222222222\", \"call_sid\":\"PNXXXXY\"}", 404));
+        when(twilioRestClient.getObjectMapper()).thenReturn(objectMapper);
+        Account.Status status = Account.Status.IN_PROGRESS;
+        AccountUpdater accountUpdater = new AccountUpdater(ACCOUNT_SID,status);
+        accountUpdater.setPauseBehavior("test");
+        Account account = accountUpdater.update(twilioRestClient);
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void testShouldHaveInValidParametersForAccountUpdater() {
+        Request mockRequest = new Request(
+                HttpMethod.POST,
+                com.twilio.rest.Domains.API.toString(),
+                "/2010-04-01/Accounts/AC222222222222222222222222222222.json"
+        );
+        mockRequest.addPostParam("PauseBehavior", "test");
+
+        ObjectMapper objectMapper = new ObjectMapper();
+        objectMapper.registerModule(new JavaTimeModule());
+        when(twilioRestClient.request(mockRequest)).thenReturn(new Response("{\"account_sid\":\"AC222222222222222222222222222222\", \"call_sid\":\"PNXXXXY\"}", 404));
+        when(twilioRestClient.getObjectMapper()).thenReturn(objectMapper);
+        Account.Status status = Account.Status.IN_PROGRESS;
+        AccountUpdater accountUpdater = new AccountUpdater(status);
+        accountUpdater.setPauseBehavior("test");
+        Account account = accountUpdater.update(twilioRestClient);
+    }
+
+    @Test
+    public void testShouldMakeValidAPICallAWSFetcher() {
+        String url = "https://api.twilio.com/v1/Credentials/AWS";
+        ObjectMapper objectMapper = new ObjectMapper();
+        objectMapper.registerModule(new JavaTimeModule());
+        String testResponse =  "{\"credentials\":[{\"sid\":\"CR12345678123456781234567812345678\", \"test_string\":\"Ahoy\", \"test_object\":{\"mms\": true, \"sms\":false, \"voice\": false, \"fax\":true}}, {\"sid\":\"CR12345678123456781234567812345678\", \"test_string\":\"Hello\", \"test_object\":{\"mms\": true, \"sms\":false, \"voice\": false, \"fax\":true}}], \"meta\": {\"url\":\"" + url + "\", \"next_page_url\":\"" + url + "?PageSize=5" + "\", \"previous_page_url\":\"" + url + "?PageSize=3" + "\", \"first_page_url\":\"" + url + "?PageSize=1" + "\", \"page_size\":4}}";
+        when(twilioRestClient.request(Mockito.any() )).thenReturn(new Response(testResponse, 200));
+        when(twilioRestClient.getObjectMapper()).thenReturn(objectMapper);
+
+        Aws resource = new AwsFetcher(ACCOUNT_SID).fetch(twilioRestClient);
+
+        assertNotNull(resource);
```

Comment:
Rather than asserting the obj to be notNull, can we assert if the response is as expected ?

Reply:
Done